Web.HtmlSanitizer/HtmlSanitizer.cs

```diff
@@ -1,460 +1,457 @@
-                // TODO: Wouldn't it be nicer is we generalized attribute rules for both checks and overrides? Would untangle code.
-                if (WhiteListMode &&
-                    !rule.SetAttributes.ContainsKey(attribute.Name) &&
-                    !rule.AttributeChecks.ContainsKey(attribute.Name) && attribute.Name != "class")
-                {
-                    attribute.Remove();
-                    return SanitizerOperation.DoNothing;
-                }
-            }
-            
-            // Do nothing else.
-            return SanitizerOperation.DoNothing;
-        }
-
-        /// <summary>
-        /// Returns an instance of the HtmlSanitizer with a HTML5 compliant rule set for documents with simple markup.
-        /// </summary>
-        /// <remarks>Strips all CSS and only allows simple links. Enfores nofollow.</remarks>
-        /// <returns></returns>
-        public static HtmlSanitizer SimpleHtml5Sanitizer()
-        {
-
-            var sanitizer = new HtmlSanitizer();
-
-            sanitizer.WhiteListMode = true;
-            sanitizer.Tag("h1").RemoveEmpty();
-            sanitizer.Tag("h2").RemoveEmpty();
-            sanitizer.Tag("h3").RemoveEmpty();
-            sanitizer.Tag("h4").RemoveEmpty();
-            sanitizer.Tag("h5").RemoveEmpty();
-            sanitizer.Tag("strong").RemoveEmpty();
-            sanitizer.Tag("b").Rename("strong").RemoveEmpty();
-            sanitizer.Tag("i").RemoveEmpty();
-            sanitizer.Tag("em");
-            sanitizer.Tag("br");
-            sanitizer.Tag("p");
-            sanitizer.Tag("div").NoAttributes(SanitizerOperation.FlattenTag);
-            sanitizer.Tag("span").RemoveEmpty();
-            sanitizer.Tag("ul");
-            sanitizer.Tag("ol")
-                .AllowAttributes("start")
-                .AllowAttributes("type")
-                .AllowAttributes("reversed");
-            sanitizer.Tag("li")
-                .AllowAttributes("value");
-            sanitizer.Tag("a").SetAttribute("target", "_blank")
-                .SetAttribute("rel", "nofollow")
-                .CheckAttributeUrl("href")
-                .RemoveEmpty()
-                .NoAttributes(SanitizerOperation.FlattenTag);
-
-            return sanitizer;
-        }
-
-        /// <summary>
-        /// Equal to the SimpleHtml5Sanitizer but allows html and body declarations.
-        /// </summary>
-        /// <returns></returns>
-        public static HtmlSanitizer SimpleHtml5DocumentSanitizer()
-        {
-
-            var sanitizer = SimpleHtml5Sanitizer();
-
-            sanitizer.Tag("html");
-            sanitizer.Tag("body");
-
-            return sanitizer;
-        }
-    }
+	internal static readonly string[] defaultAllowedUriSchemes = new[] { "http", "https", "mailto", "tel" };
+
+	private readonly CssWhitelistAttributeSanitizer _cssAttributeSanitizer;
+
+	/// <summary>Initializes a new instance of the <see cref="HtmlSanitizer"/> class.</summary>
+	[Obsolete]

```

Comment:
Is there a reason for marking the constructor `[Obsolete]`? I see that the compiler doesn't complain when constructing the class using the `new()` expression, but I do not see why we should discourage anyone from calling the constructor?